pkg/diagnose/diagnosis/loader.go

```diff
@@ -35,10 +34,7 @@ func RegisterMetadataAvail(name string, d MetadataAvailDiagnose) {
 // Diagnose (all subcommand)
 
 // Diagnose interface function
-type Diagnose func(Config, sender.DiagnoseSenderManager) []Diagnosis
-
-// Catalog is a global list of registered Diagnose functions
-var Catalog = make([]Suite, 0)
+type Diagnose func() []Diagnosis
```

Comment:
I wonder how config is passed down to all diagnose functions

Reply:
The config is captured in [getSuites](https://github.com/DataDog/datadog-agent/pull/22692/files#diff-7490fc6d3c928262876d9ad313837f636bca99ba772e5c448ad4e9619dddd1e0R411) only if a check needs it.